diff --git a/src/w32-gettext.c b/src/w32-gettext.c
index 42aa01c..bf085cd 100644
--- a/src/w32-gettext.c
+++ b/src/w32-gettext.c
@@ -1233,7 +1233,15 @@ load_domain (const char *filename)
   if (fh == INVALID_HANDLE_VALUE)
     return NULL;
 
+#if !WINAPI_FAMILY_PARTITION (WINAPI_PARTITION_DESKTOP)
+  FILE_STANDARD_INFO fStdInfo;
+  if (GetFileInformationByHandleEx(fh, FileStandardInfo, (LPVOID)&fStdInfo, sizeof(fStdInfo)))
+    size = fStdInfo.EndOfFile.LowPart;
+  else
+    size = INVALID_FILE_SIZE;
+#else
   size = GetFileSize (fh, NULL);
+#endif
   if (size == INVALID_FILE_SIZE)
     {
       CloseHandle (fh);
