diff --git a/compat/compat.h b/compat/compat.h
index eadfae8..014298c 100644
--- a/compat/compat.h
+++ b/compat/compat.h
@@ -31,6 +31,10 @@
 #ifndef BDSM_COMPAT_H
 # define BDSM_COMPAT_H
 
+#ifdef _WIN32
+# define _CRT_RAND_S    /* needed before including stdlib.h !!! */
+#endif
+
 #include <stdlib.h>
 #if !defined HAVE_STRLCPY && !defined HAVE_LIBBSD
 size_t strlcpy(char *dst, const char *src, size_t siz);
diff --git a/src/smb_ntlm.c b/src/smb_ntlm.c
index 998afd3..c55d169 100644
--- a/src/smb_ntlm.c
+++ b/src/smb_ntlm.c
@@ -32,10 +32,6 @@
 # include "config.h"
 #endif
 
-#ifdef _WIN32
-# define _CRT_RAND_S
-#endif
-
 #include <assert.h>
 #include <ctype.h>
 #include <wctype.h>
