diff --git a/src/lmathlib.c b/src/lmathlib.c
index e0c61a1..7cfac2b 100644
--- a/src/lmathlib.c
+++ b/src/lmathlib.c
@@ -173,7 +173,7 @@ static int math_log (lua_State *L) {
     res = l_mathop(log)(x);
   else {
     lua_Number base = luaL_checknumber(L, 2);
-#if !defined(LUA_USE_C89)
+#if !defined(LUA_USE_C89) && !(defined(__ANDROID__) && (__ANDROID_API__ < 18))
     if (base == l_mathop(2.0))
       res = l_mathop(log2)(x);
     else
