diff --git a/Makefile.am b/Makefile.am
index 77355d1a..737f32ad 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -41,8 +41,8 @@ doc =
 endif
 
 
-DIST_SUBDIRS = m4 compat mpi cipher random src doc tests
-SUBDIRS =         compat mpi cipher random src $(doc) tests
+DIST_SUBDIRS = m4 compat mpi cipher random src doc
+SUBDIRS =         compat mpi cipher random src $(doc)
 
 EXTRA_DIST = autogen.sh autogen.rc README.GIT LICENSES      	           \
              ChangeLog-2011 build-aux/ChangeLog-2011 doc/ChangeLog-2011    \
