diff --git a/CMakeLists.txt b/CMakeLists.txt
index de7f4525..6fd04bf4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -301,6 +301,15 @@ if (UPNP_ENABLE_OPEN_SSL)
 	set (OPENSSL_LIBS "-lssl")
 endif()
 
+if (WIN32 OR CMAKE_SYSTEM_NAME STREQUAL "Linux")
+    set (VLC_FORCED_LIBS "-pthread -lpthread")
+
+    # Force network libs.
+    if (WIN32)
+        set (VLC_FORCED_LIBS "${VLC_FORCED_LIBS} -liphlpapi -lws2_32")
+    endif()
+endif()
+
 configure_file (${CMAKE_CURRENT_SOURCE_DIR}/libupnp.pc.in ${CMAKE_CURRENT_BINARY_DIR}/libupnp.pc @ONLY)
 
 install (FILES ${CMAKE_CURRENT_BINARY_DIR}/libupnp.pc
diff --git a/libupnp.pc.in b/libupnp.pc.in
index 8c6fc22f..4c8b9f79 100644
--- a/libupnp.pc.in
+++ b/libupnp.pc.in
@@ -6,7 +6,7 @@ includedir=@includedir@
 Name: libupnp
 Description: Linux SDK for UPnP Devices
 Version: @VERSION@
-Libs: @PTHREAD_CFLAGS@ @PTHREAD_LIBS@ -L${libdir} -lupnp -lixml
+Libs: @PTHREAD_CFLAGS@ @PTHREAD_LIBS@ -L${libdir} -lupnp -lixml @VLC_FORCED_LIBS@
 Libs.private: @OPENSSL_LIBS@
 Cflags: @PTHREAD_CFLAGS@ -I${includedir}/upnp
 
