diff --git a/CMakeLists.txt b/CMakeLists.txt
index 33f684f5..ea79c4aa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -52,6 +52,9 @@ function(configure_pkg_config_file pkg_config_file_in)
     set(libdir ${CMAKE_INSTALL_FULL_LIBDIR})
     set(includedir ${CMAKE_INSTALL_FULL_INCLUDEDIR})
     set(VERSION ${PROJECT_VERSION})
+    if(HAVE_LIBM)
+        set(VORBIS_LIBS "-lm")
+    endif()
     string(REPLACE ".in" "" pkg_config_file ${pkg_config_file_in})
     configure_file(${pkg_config_file_in} ${pkg_config_file} @ONLY)
 endfunction()
