diff --git a/src/MXF.h b/src/MXF.h
index 8ce77e1..32422b2 100755
--- a/src/MXF.h
+++ b/src/MXF.h
@@ -263,30 +263,6 @@ namespace ASDCP
 	  const PropertyType& const_get() const { return m_property; }
 	};
 
-      // wrapper object manages optional properties
-      template <class PropertyType>
-	class optional_container_property
-	{
-	  PropertyType m_property;
-
-	public:
-	  optional_container_property() {}
-	optional_container_property(const PropertyType& value) : m_property(value) {}
-	  const optional_container_property<PropertyType>& operator=(const PropertyType& rhs) {
-	    this->Copy(rhs.m_property);
-	    return *this;
-	  }
-
-	  bool operator==(const PropertyType& rhs) const { return this->m_property == rhs; }
-	  bool operator==(const optional_property<PropertyType>& rhs) const { return this->m_property == rhs.m_property; }
-	  operator PropertyType&() { return this->m_property; }
-	  void set(const PropertyType& rhs) { this->m_property = rhs; }
-	  void reset(const PropertyType& rhs) { this->clear(); }
-	  bool empty() const { return ! this->m_property.HasValue(); }
-	  PropertyType& get() { return m_property; }
-	  const PropertyType& const_get() const { return m_property; }
-	};
-
       // base class of all metadata objects
       //
       class InterchangeObject : public ASDCP::KLVPacket
