diff --git a/fxc2.cpp b/fxc2.cpp
index 17b0ccb..dbc5728 100755
--- a/fxc2.cpp
+++ b/fxc2.cpp
@@ -306,10 +306,10 @@ int main(int argc, char* argv[])
   if (FAILED(hr)) {
    if (errors) {
     char* error = (char*)errors->GetBufferPointer();
-    printf("Got an error (%i) while compiling:\n%s\n", hr, error);
+    printf("Got an error (0x%lX) while compiling: '%s'\n%s\n", hr, inputFile, error);
     errors->Release();
    } else {
-     printf("Got an error (%i) while compiling, but no error message from the function.\n", hr);
+     printf("Got an error (0x%lX) while compiling '%s', but no error message from the function.\n", hr, inputFile);
 
      LPSTR messageBuffer = nullptr;
      size_t size = FormatMessageA(FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
