diff --git a/fxc2.cpp b/fxc2.cpp
index 5b8c86c..205fa16 100755
--- a/fxc2.cpp
+++ b/fxc2.cpp
@@ -317,7 +317,7 @@ int main(int argc, char* argv[])
     FILE* f;
     errno_t err = fopen_s(&f, outputFile, "w");
 
-    fprintf(f, "const signed char %s[] =\n{\n", variableName);
+    fprintf(f, "const BYTE %s[] =\n{\n", variableName);
     for (i = 0; i < len; i++) {
      fprintf(f, "%4i", outString[i]);
      if (i != len - 1)
@@ -326,7 +326,7 @@ int main(int argc, char* argv[])
        fprintf(f, "\n");
     }
 
-    fprintf(f, "\n};\n");
+    fprintf(f, "\n};");
     fclose(f);
 
     if(verbose) {
