diff --git a/source/CMakeLists.txt b/source/CMakeLists.txt
index 33b6523f1..0ea2487c7 100644
--- a/source/CMakeLists.txt
+++ b/source/CMakeLists.txt
@@ -713,7 +713,7 @@ if(ENABLE_CLI)
         if(WIN32 OR NOT ENABLE_SHARED OR INTEL_CXX)
             # The CLI cannot link to the shared library on Windows, it
             # requires internal APIs not exported from the DLL
-            target_link_libraries(cli x265-static ${PLATFORM_LIBS})
+            target_link_libraries(cli x265-static ${PLATFORM_LIBS} ${CMAKE_DL_LIBS})
         else()
             target_link_libraries(cli x265-shared ${PLATFORM_LIBS})
         endif()
