diff --git a/source/encoder/api.cpp b/source/encoder/api.cpp
index 10031d109..46d84a75c 100644
--- a/source/encoder/api.cpp
+++ b/source/encoder/api.cpp
@@ -720,10 +720,10 @@ const x265_api* x265_api_get(int bitDepth)
             g_recursion++;
 
 #if _WIN32
-        HMODULE h = LoadLibraryA(libname);
+        HMODULE h = LoadLibraryExA(libname, NULL, LOAD_LIBRARY_SEARCH_DEFAULT_DIRS);
         if (!h)
         {
-            h = LoadLibraryA(multilibname);
+            h = LoadLibraryExA(multilibname, NULL, LOAD_LIBRARY_SEARCH_DEFAULT_DIRS);
             reqDepth = bitDepth;
         }
         if (h)
@@ -813,10 +813,10 @@ const x265_api* x265_api_query(int bitDepth, int apiVersion, int* err)
             g_recursion++;
 
 #if _WIN32
-        HMODULE h = LoadLibraryA(libname);
+        HMODULE h = LoadLibraryExA(libname, NULL, LOAD_LIBRARY_SEARCH_DEFAULT_DIRS);
         if (!h)
         {
-            h = LoadLibraryA(multilibname);
+            h = LoadLibraryExA(multilibname, NULL, LOAD_LIBRARY_SEARCH_DEFAULT_DIRS);
             reqDepth = bitDepth;
         }
         if (h)
