diff --git a/source/common/cpu.cpp b/source/common/cpu.cpp
index ae0907890..99cdc3e92 100644
--- a/source/common/cpu.cpp
+++ b/source/common/cpu.cpp
@@ -128,6 +128,11 @@ const cpu_name_t cpu_names[] =
     { "", 0 },
 };
 
+bool detect512()
+{
+    return(enable512);
+}
+
 #if X265_ARCH_X86
 
 extern "C" {
@@ -141,10 +146,6 @@ uint64_t PFX(cpu_xgetbv)(int xcr);
 #pragma warning(disable: 4309) // truncation of constant value
 #endif
 
-bool detect512()
-{
-    return(enable512);
-}
 
 uint32_t cpu_detect(bool benableavx512 )
 {
