diff --git a/source/CMakeLists.txt b/source/CMakeLists.txt
index a6ffcc6fd..7a47a7bef 100755
--- a/source/CMakeLists.txt
+++ b/source/CMakeLists.txt
@@ -957,7 +957,7 @@ if(X265_LATEST_TAG OR NOT GIT_FOUND)
     endforeach()
     if(PLIBLIST)
         # blacklist of libraries that should not be in Libs.private
-        list(REMOVE_ITEM PLIBLIST "-lc" "-lpthread" "-lmingwex" "-lmingwthrd"
+        list(REMOVE_ITEM PLIBLIST "-lc" "-lmingwex" "-lmingwthrd"
             "-lmingw32" "-lmoldname" "-lmsvcrt" "-ladvapi32" "-lshell32"
             "-luser32" "-lkernel32")
         string(REPLACE ";" " " PRIVATE_LIBS "${PLIBLIST}")
