diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7403e1891..5c5f4e5ce 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -336,7 +336,7 @@ function(get_pkgconfig_deps NAME OUT)
     list(REMOVE_ITEM link_libs "${OPENJPEG_LIBRARY_NAME}")
     set(deps "")
     foreach(item IN LISTS link_libs)
-      if(item STREQUAL "-pthread")
+      if(item STREQUAL "-pthread" OR item STREQUAL "-lpthread")
         string(APPEND deps " ${item}")
       else()
         string(APPEND deps " -l${item}")
