diff --git a/src/qml/common/qv4compileddata.cpp b/src/qml/common/qv4compileddata.cpp
index 803739e27c..1af7319517 100644
--- a/src/qml/common/qv4compileddata.cpp
+++ b/src/qml/common/qv4compileddata.cpp
@@ -68,20 +68,6 @@ bool Unit::verifyHeader(QDateTime expectedSourceTimeStamp, QString *errorString)
         }
     }
 
-#if defined(QML_COMPILE_HASH) && defined(QML_COMPILE_HASH_LENGTH) && QML_COMPILE_HASH_LENGTH > 0
-    if (qstrncmp(qml_compile_hash, libraryVersionHash, QML_COMPILE_HASH_LENGTH) != 0) {
-        *errorString = QStringLiteral("QML compile hashes don't match. Found %1 expected %2")
-                               .arg(QString::fromLatin1(
-                                            QByteArray(libraryVersionHash, QML_COMPILE_HASH_LENGTH)
-                                                    .toPercentEncoding()),
-                                    QString::fromLatin1(
-                                            QByteArray(qml_compile_hash, QML_COMPILE_HASH_LENGTH)
-                                                    .toPercentEncoding()));
-        return false;
-    }
-#else
-#error "QML_COMPILE_HASH must be defined for the build of QtDeclarative to ensure version checking for cache files"
-#endif
     return true;
 }
 
