diff --git a/src/qml/common/qv4compileddata.cpp b/src/qml/common/qv4compileddata.cpp
index 9dee91f713..0ea3e2912b 100644
--- a/src/qml/common/qv4compileddata.cpp
+++ b/src/qml/common/qv4compileddata.cpp
@@ -49,7 +49,7 @@ bool Unit::verifyHeader(QDateTime expectedSourceTimeStamp, QString *errorString)
         return false;
     }
 
-    if (qtVersion != quint32(QT_VERSION)) {
+    if ((qtVersion >> 8) != (quint32(QT_VERSION) >> 8)) {
         *errorString = QString::fromUtf8("Qt version mismatch. Found %1 expected %2")
                                .arg(qtVersion, 0, 16).arg(QT_VERSION, 0, 16);
         return false;
