diff --git a/cmake/GeneratePkgConfig.cmake b/cmake/GeneratePkgConfig.cmake
index ae9add8..aebf2c1 100644
--- a/cmake/GeneratePkgConfig.cmake
+++ b/cmake/GeneratePkgConfig.cmake
@@ -30,7 +30,9 @@ function(generate_pkg_config_pc_file TARGET TEMPLATE template OUTPUT output)
     if(NOT BUILD_SHARED_LIBS)
         if(CMAKE_CXX_IMPLICIT_LINK_LIBRARIES)
             foreach(IMPLICIT_LIB ${CMAKE_CXX_IMPLICIT_LINK_LIBRARIES})
-                if (IMPLICIT_LIB MATCHES "-l.*")
+                if(IMPLICIT_LIB MATCHES "-l:libunwind.a") # android toolchain
+                    list(APPEND LIBS_LIST "-lunwind")
+                elseif (IMPLICIT_LIB MATCHES "(-l|/).*")
                     list(APPEND LIBS_LIST "${IMPLICIT_LIB}")
                 else()
                     list(APPEND LIBS_LIST "-l${IMPLICIT_LIB}")
