diff --git a/src/dvbpsi.c b/src/dvbpsi.c
index a11d70a..0c8c440 100644
--- a/src/dvbpsi.c
+++ b/src/dvbpsi.c
@@ -315,7 +315,11 @@ bool dvbpsi_packet_push(dvbpsi_t *p_dvbpsi, uint8_t* p_data)
 
     /* Skip the adaptation_field if present */
     if (p_data[3] & 0x20)
+    {
         p_payload_pos = p_data + 5 + p_data[4];
+	if(p_payload_pos >= p_data_end)
+	    return false;
+    }
     else
         p_payload_pos = p_data + 4;
 
