diff --git a/src/dvbpsi.c b/src/dvbpsi.c
index 0c8c440..6b5641a 100644
--- a/src/dvbpsi.c
+++ b/src/dvbpsi.c
@@ -328,6 +328,8 @@ bool dvbpsi_packet_push(dvbpsi_t *p_dvbpsi, uint8_t* p_data)
     {
         p_new_pos = p_payload_pos + *p_payload_pos + 1;
         p_payload_pos += 1;
+	if(p_payload_pos >= p_data_end || p_new_pos >= p_data_end)
+	    return false;
     }
 
     p_section = p_decoder->p_current_section;
