--- a/src/libbluray/bdj/meson.build
+++ b/src/libbluray/bdj/meson.build
@@ -10,9 +10,7 @@
     if get_option('java9') and bdj_type == 'j2se'
         if javac_version.version_compare('>= 18.0')
             java_src_version = '1.8'
-        elif javac_version.version_compare('>=12.0')
-            java_src_version = '1.7'
         elif javac_version.version_compare('>=9.0')
-            java_src_version = '1.6'
+            java_src_version = '1.8'
         else
             java_src_version = '1.5' #maybe java_version_bdj=1.4 in that case?
         endif
